message_logs_to_sql_insert.py

```diff
@@ -0,0 +1,144 @@
+#--------------------------
+# CAMPUS MESSAGING LOG PARSER
+#
+# This script processes all the *.log files in
+# its directory, and prepares an SQL file
+# which will insert values into
+# UCOPRerports.[Message Files] and UCOPRerports.[Messages]
+#--------------------------
+
+import glob
+import sys
+from datetime import datetime
+
+# Set to true to output email address CSV
+output_csv = False;
+
+# Gets all the log files in the folder where the python script is run
+print("Log files to process: ", glob.glob("*.log"), "\n") 
+log_file_array = glob.glob("*.log")
+
+for log_file in log_file_array:
+	# File passed as an argument
+	# fname = sys.argv[1]
+	fname = log_file
+
+	# Open file / exception
+	try:
+		fhand = open(fname)
+	except:
+		print('File cannot be opened:',fhand)
+		exit()
+
+	#--------------------------
+	# Check for "send mode" text.
+	send_mode_check = False
+	for line in fhand:
+		if (send_mode_check == False) and (line.find("SEND MODE") != -1):
+			send_mode_check = True
+			break
+
+	if (send_mode_check == False):
+		print("Not processed: " + fname + " - File does not contain SEND MODE.")
+		continue
+	else:
+		print("Processing:\n" + fname +"\nFile contains SEND MODE.")
+
+
+	#--------------------------
+	# Pull the emails and set the date
+	email_addresses = []
+	email_datetime = None
+
+	# Find the sending email message
+	for line in fhand:
+		if line != None and "sending message to" in line:
+
+			# add the email address to the list
+			email_ad = (line.split(' ')[3]).strip('\n')
+			email_addresses.append(email_ad)
+
+		# Find the date
+		if line != None and email_datetime == None and "Date: " in line:
+
+			# Split line to get just the date
+			email_date = ""
+			line_split = line.split(' ')
+			for i in range(2, 5):
+				email_date += line_split[i]
+
+			# convert to date object
+			email_datetime = datetime.strptime(email_date, '%d%b%Y')
+
+
+	#--------------------------
+	# Output files
+
+	# Message Files SQL
+	sql_file = open (fname + "-insert.sql", "a")
+	table_name = "UCOPReports.[Message Files]"
+	column_names = "([File Name],[Email Date])"
+	sql_file.write("INSERT INTO " + table_name + "\n")
```

Comment:
The script processes all the log file. Great. I'm looking for where it checks if a log file has already been processed?

Reply:
It doesn't check [Message Files] for duplicates -- this should be added before deploying (just to be on the safe side)